app/src/main/java/org/p2p/wallet/newsend/SendFeeRelayerManager.kt

```diff
@@ -314,10 +326,27 @@ class SendFeeRelayerManager(
             Timber.e(e, "Error calculating fees")
             handleError(FeesCalculationError)
             null
-        } ?: return
-        val fee = buildSolanaFee(newFeePayer, sourceToken, feeRelayerFee)
-        validateFunds(sourceToken, fee, inputAmount)
-        currentState = UpdateFee(fee, feeLimitInfo)
+        }
+
+        when (feeState) {
+            is FeeCalculationState.NoFees -> {
+                currentState = UpdateFee(solanaFee = null, feeLimitInfo = feeLimitInfo)
+            }
+            is FeeCalculationState.PoolsNotFound -> {
+                val solanaFee = buildSolanaFee(solToken, sourceToken, feeState.feeInSol)
+                currentState = UpdateFee(solanaFee = solanaFee, feeLimitInfo = feeLimitInfo)
+                sendInteractor.setFeePayerToken(solToken)
+            }
+            is FeeCalculationState.Success -> {
+                val fee = buildSolanaFee(newFeePayer, sourceToken, feeState.fee)
+                validateFunds(sourceToken, fee, inputAmount)
+                currentState = UpdateFee(fee, feeLimitInfo)
+            }
+            is FeeCalculationState.Error -> {
+                handleError(FeesCalculationError)
+            }
+            else -> Unit
```

Comment:
i would suggest not using else when working with Enums, if we add new state - compiler won't warn you that you haven't handled it

Reply:
compiler warning, I had to